### PR TITLE
feat: implement network interception provide response

### PR DESCRIPTION
**Pull request by thiagowfx**
Bug: #644

### CODE REVIEW

**Comment by thiagowfx** [comment]
This is now ready for review

File: src/bidiMapper/domains/network/NetworkProcessor.ts [hunk 1]
```diff
@@ -163,8 +163,37 @@ export class NetworkProcessor {
...
+
+    // TODO: Set / expand.
+    // ; Step 10. cookies
+    // ; Step 11. credentials
+    const request = this.#networkStorage.getRequest(networkId);
+    assert(request, `Network request with ID ${networkId} doesn't exist`);
+
+    await request.provideResponse(
+      fetchId,
+      statusCode!, // TODO: What if the status code is not present?
```
Comment (sadym-chromium) [inline src/bidiMapper/domains/network/NetworkProcessor.ts:188]: Provide one from the request, it should be there.

Comment (thiagowfx) [reply src/bidiMapper/domains/network/NetworkProcessor.ts]: We still need to provide a fallback in case it is undefined. I am using 200. Is that OK?

**Review by sadym-chromium — APPROVED** [review]
One comment regarding the test.

File: tests/network/test_provide_response.py [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+                    },
+                }, ],
+                "url": example_url,
+                "status": 200,  # TODO: Update status code to 501?
+                "statusText": "Remember to drink water",
+            }),
+            "timestamp": ANY_TIMESTAMP,
+        },
+        "type": "event",
+    }
```
[the remark this answers is not in the document]
Comment (thiagowfx) [reply tests/network/test_provide_response.py:387]: Response contains `'bodySize': 0,`. This seems incorrect.

**Pull request merged**